File: app/src/main/java/fansirsqi/xposed/sesame/entity/MemberBenefit.java
package fansirsqi.xposed.sesame.entity;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import java.util.Set;

import fansirsqi.xposed.sesame.util.Maps.IdMapManager;
import fansirsqi.xposed.sesame.util.Maps.MemberBenefitsMap;

public class MemberBenefit extends MapperEntity {

    public MemberBenefit(String i, String n) {
        id = i;
        name = n;
    }

    public static List<MemberBenefit> getList() {
        List<MemberBenefit> list = new ArrayList<>();
        Map<String, String> idSet = IdMapManager.getInstance(MemberBenefitsMap.class).getMap();
        for (Map.Entry<String, String> entry: idSet.entrySet()) {
            list.add(new MemberBenefit(entry.getKey(), entry.getValue()));
        }
        return list;
    }
}

File: app/src/main/java/fansirsqi/xposed/sesame/entity/MemberBenefit.kt
package fansirsqi.xposed.sesame.entity

import fansirsqi.xposed.sesame.util.maps.IdMapManager
import fansirsqi.xposed.sesame.util.maps.MemberBenefitsMap

class MemberBenefit(i: String, n: String) : MapperEntity() {

    init {
        id = i
        name = n
    }

    companion object {
        fun getList(): List<MemberBenefit> {
            return IdMapManager.getInstance(MemberBenefitsMap::class.java).map
                .map { (key, value) -> MemberBenefit(key, value) }
        }
    }
}